yt_dlp/YoutubeDL.py

```diff
@@ -625,7 +625,7 @@ def __init__(self, params=None, auto_init=True):
                     'Overwriting params from "color" with "no_color"')
             self.params['color'] = 'no_color'
 
-        term_allow_color = os.environ.get('TERM', '').lower() != 'dumb'
+        term_allow_color = os.getenv('TERM', '').lower() != 'dumb' and not os.getenv('NO_COLOR')
 
         def process_color_policy(stream):
             stream_name = {sys.stdout: 'stdout', sys.stderr: 'stderr'}[stream]
```

Comment:
After thinking about it for some more time, id say we should do this:
```py
        term_allow_color = os.getenv('TERM', '').lower() != 'dumb'
        no_color = bool(os.getenv('NO_COLOR'))

        def process_color_policy(stream):
            stream_name = {sys.stdout: 'stdout', sys.stderr: 'stderr'}[stream]
            policy = traverse_obj(self.params, ('color', (stream_name, None), {str}), get_all=False)
            if policy in ('auto', None):
                if term_allow_color and supports_terminal_sequences(stream):
                    return 'no_color' if no_color else True
                return False
```
This means you can override the environment variable in code and cli, as well as only disabling colored output, not removing all terminal sequences

Reply:
That looks good to me.

Reply:
Feel free to commit that and we can merge it

Reply:
> Feel free to commit that and we can merge it
Done as of fe89d6aa4cd68019e427cd0a020882587f0928b9.